pkg/api/login_oauth.go

```diff
@@ -217,9 +217,12 @@ func (hs *HTTPServer) OAuthLogin(ctx *m.ReqContext) {
 	metrics.MApiLoginOAuth.Inc()
 
 	if redirectTo, _ := url.QueryUnescape(ctx.GetCookie("redirect_to")); len(redirectTo) > 0 {
-		middleware.DeleteCookie(ctx.Resp, "redirect_to", hs.cookieOptionsFromCfg)
-		ctx.Redirect(redirectTo)
-		return
+		if err := hs.validateRedirectTo(redirectTo); err == nil {
+			middleware.DeleteCookie(ctx.Resp, "redirect_to", hs.cookieOptionsFromCfg)
```

Comment:
happen to understand this enough to add a comment explaining why the cookie is deleted?

Reply:
the cookie is no longer needed since it held the redirect_to location.  this code path is going to send the browser to the new location, and should not retain the redirect_to cookie (could end up in a redirect loop, but mainly it is no longer needed).

Reply:
exactly; the cookie is deleted because it's has fulfilled its purpose.